force-app/main/default/classes/TriggerFactory.cls

```diff
@@ -0,0 +1,41 @@
+public class TriggerFactory {
+	public class TriggerException extends Exception {
+	}
+	public static void execute(SObjectType sobjectType) {
+		ITriggerHandler handler = createHandler(sobjectType);
+		handler.disableTrigger();
+		handler.enableTrigger();
+		if (!handler.isTriggerEnabled()) {
+			return;
+		}
+		handleOperation(handler);
+	}
+
+	private static void handleOperation(ITriggerHandler handler) {
+		if (Trigger.isUpdate) {
```

Comment:
Add decision for all operation even if you don't have it now in trigger: for delete, insert for example
Also to make it look better we can use switch and trigger opertaion like:
switch on Trigger.OperationType {
            when BEFORE_INSERT {
            }
 }
 
 https://developer.salesforce.com/blogs/2018/05/summer18-rethink-trigger-logic-with-apex-switch (Switch and enum in triggers part)